src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/TagHelperBlockRewriter.cs

```diff
@@ -769,6 +785,32 @@ public override SyntaxNode VisitMarkupEphemeralTextLiteral(MarkupEphemeralTextLi
                 return value.WithSpanContext(node.GetSpanContext());
             }
 
+            // Being collapsed represents that a block contains several identical looking markup literal attribute values. This can be the case
+            // when a user has written something like: @onclick="() => SomeMethod()"
+            // In that case there would be 3 children:
+            //   - ()
+            //   -  =>
+            //   -  SomeMethod()
+            // There are 3 children because the Razor parser separates attribute values based on whitespace.
+            private static bool CanBeCollapsed(GenericBlockSyntax node)
```

Comment:
So is this logic basically - does it contain any `@` or comment? I see you're using an allow-list here which makes me wonder if more things need to be on that list.

Reply:
> So is this logic basically - does it contain any `@` or comment?

Nope, I don't currently handle that here because it's handled properly by other call paths.

> I see you're using an allow-list here which makes me wonder if more things need to be on that list.

Definitely could, I tried to make this as targeted as possible while still not writing dead code to not harm other scenarios.

Reply:
> So is this logic basically - does it contain any `@` or comment?

Nope, I don't currently handle that here because it's handled properly by other call paths.

> I see you're using an allow-list here which makes me wonder if more things need to be on that list.

Definitely could, I tried to make this as targeted as possible while still not writing dead code to not harm other scenarios.